text/0000-buildpack-registry-search-api.md

```diff
@@ -0,0 +1,172 @@
+  ```json
+  [
+    {
+      "ns":"projectriff",
+      "name":"command-function",
+      "version":"1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"
+    },
+    {
+      "ns":"projectriff",
+      "name":"java-function",
+      "version":"1.4.1",
+      "yanked":false,"addr":"gcr.io/projectriff/java-function@sha256:5eabea8f7b2c09074ec196fe0c321006fb5ad8f282cc918520286d8a0007196f"},
+    {
+      "ns":"projectriff",
+      "name":"node-function",
+      "version":"1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/node-function@sha256:194298b826c15bb079c59aed99968d7678a6e1f7a882c9d7f61811e0990717ba"},
+    {
+      "ns":"projectriff",
+      "name":"streaming-http-adapter",
+      "version":"1.4.0",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/streaming-http-adapter@sha256:b202d9ec203e882ee7e3c599d9e867617f909c8b6123e4ce942af47db6e58c45"
+    }
+  ]
+  ```
+
+- **GET /buildpacks/:id**
+
+  Retrieves metadata for a specific buildpack.  This will contain more detailed data.
+  ```
+  $ curl https://glacial-plateau-95350.herokuapp.com/buildpacks/1
+  ```
+  ```json
+  {
+      "description": "The Command Function Buildpack is a Cloud Native Buildpack V3 that provides riff Command Function Invoker to functions",
+      "license": "MIT",
+      "ns":"projectriff",
+      "name":"command-function",
+      "latest_version": "1.4.1",
+      "versions": "/buildpacks/1/versions",
+      "weekly_downloads": 8993,
+      "issues": 44,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"
+  }
+  ```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In the initial implementation, we can take advantage of the existing [registry](https://github.com/buildpacks/registry-index) repository to pull buildpack data from.  Pack already performs a clone/pull on this repo for its registry cache.  To keep ensure up-to-date buildpack data, we'd have a polling loop that performs a `git pull`.  This polling would take place in a separate background process.  
+
+Each time the local repository has been updated via `git pull`, buildapack data will be processed/normalized into a single JSON object, where plain text searches can be used against it.   Fields in this JSON object will be re-indexed as searchable fields, which will be compared against the search text during the retrieval algorithm.
+
+In addition, a server process will expose endpoints and handle incoming request for the GET endpoints mentioned earlier.  For search requests, the text will be extracted from the `query` query parameter and used to search against the normalized buildpacks index.   Results will be added to the server response, as a list of JSON objects.
+
+*Note:  The Distribution Team maintainers can manage this service, and could even set-up a pager (but only best effort, and during working hours).
```

Comment:
We should probably open it up to verified project contributors as well so there is less stress or demand on just a few and also have a bit more knowledge transfer.

Reply:
Sounds good to me!  I just didn't want to volunteer others to participate, but if others are willing, the more the merrier.